hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -346,6 +349,19 @@ public synchronized void scan(Option<List<String>> keys) {
     }
   }
 
+  /**
+   * Check whether the current instants are valid
+   *
+   * @param instantTime
+   * @return
+   */
+  private boolean checkIfValidCommit(String instantTime) {
+    HoodieTimeline deltaCommitTimeline = this.hoodieTableMetaClient.getActiveTimeline().getDeltaCommitTimeline().filterCompletedInstants();
+    return deltaCommitTimeline.containsInstant(instantTime) ||
+        (deltaCommitTimeline.isBeforeTimelineStarts(instantTime) &&
+            this.hoodieTableMetaClient.getArchivedTimeline().getDeltaCommitTimeline().filterCompletedInstants().containsOrBeforeTimelineStarts(instantTime));
```

Comment:
shouldn't we only do .contains() and not before with archived timeline.

Reply:
we should compare the instantTime with active timeline and archived timeline.

not just active timeline

Reply:
Why we compare/check it for each block, that is not efficient.